### ISSUE TITLE
Wrong SHA on CommitStatusUpdater with pipeline's lib

### ISSUE THREAD

**Issue by lalmeras** [jenkinsci/gitlab-plugin#553]
## Issue

On a project build with a Jenkinsfile and pipeline's library, SHA used by CommitStatusUpdater is buggy: it's the pipeline's library sha, not the project SHA, when triggered manually or by cron.

### Context
- **Gitlab plugin version**: 1.4.5
- **Gitlab version**: 8.16.8
- **Jenkins version**: 2.46.2
- **Job type**: Pipeline with pipeline library

### Workaround, possible fix

I find that other plugins encounter the same issue with pipeline's library; I find that bug https://issues.jenkins-ci.org/browse/JENKINS-42809 which is fixed by modifying revision lookup by SCMRevisionAction in place of BuildData. I also observe the same bug with github-plugin https://issues.jenkins-ci.org/browse/JENKINS-40150

**Comment by lalmeras** [comment]
After more research, I find that CommitStatusUpdater#retrieveGitlabProjectIds is also problematic, as it retrieves scm remote url from BuildData. For my case, it also points toward pipeline's library repository instead of project repository.

As SCMRevisionAction does not provide remote information, I don't know how to fix this.

**Comment by lalmeras** [comment]
I post on https://issues.jenkins-ci.org/browse/JENKINS-42809 as it seems to me that BuildData's behavior change (it now exposes pipeline's lib information instead of project information) may need to be reworked or explain before we can fix this issue.

If anyone could confirm or refute my analysis ?

**Comment by omehegan** [comment]
This is a duplicate of #523. We have guidance on how to fix it, hopefully a PR will be forthcoming soon.

**Comment by omehegan** [comment]
This was released today in version 1.4.6 of the plugin.
